src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs

```diff
@@ -41,6 +41,14 @@ protected override IEnumerator<RavenEtlItem> ConvertTombstonesEnumerator(IEnumer
             return new TombstonesToRavenEtlItems(tombstones, collection);
         }
 
+        protected override bool ShouldTrackAttachmentTombstones()
+        {
+            // if script isn't empty we relay on addAttachment() calls and detect that attachments needs be deleted
+            // when this call gets an attachment reference marked as null ($attachment/{attachment-name}/$null)
```

Comment:
What happens if they do something like:

```
for(var attachment in this["@metadata"]["@attachments"])
{
   if(attachment.Name.endsWith(".png"))
       addAttachment(...);
}
```

In this manner, you don't _see_ that there are attachments to be deleted on the other side.

Reply:
Correct. Also it wound't work for renaming attachments. It means we always have to track them.

Reply:
So, we need to delete all the other attachments that exists on destination when we send it, no?

Reply:
True, that would make it easier.